src/foam/nanos/crunch/connection/CapabilityPayloadDAO.js

```diff
@@ -211,6 +209,17 @@ foam.CLASS({
                 }
                 catch (IllegalStateException | IllegalArgumentException ie) {
                   validationErrors.put(key, ie.getMessage());
+                } catch (ValidationException ve) {
+                  validationErrors.put(key, ve.getErrorMessage());
+                } catch (CompoundException ce) {
+                  for ( var t : ce.getExceptions() ) {
+                    if ( t instanceof ValidationException ) {
+                      var ve = (ValidationException) t;
+                      validationErrors.put(
+                        String.format("%s[%s]", key, ve.getPropName()),
```

Comment:
With our other future exception work, it would be nice to just capture keys a propertyinfos, then build exceptions with those.